lib/semantics/mod-file.cpp

```diff
@@ -675,27 +669,29 @@ static bool FileContentsMatch(const std::string &path,
   if (GetFileSize(path) != hsize + csize) {
     return false;
   }
-  int fd{open(path.c_str(), O_RDONLY)};
-  if (fd < 0) {
+  namespace fs = llvm::sys::fs;
+  auto efile =
+      fs::openNativeFile(path, fs::CD_OpenExisting, fs::FA_Read, fs::OF_None);
+  if (!efile) {
     return false;
   }
+  fs::file_t fd = efile.get();
   constexpr std::size_t bufSize{4096};
-  std::string buffer(bufSize, '\0');
-  if (read(fd, &buffer[0], hsize) != static_cast<ssize_t>(hsize) ||
+  std::vector<char> buffer(bufSize, '\0');
+  if (auto er = fs::readNativeFile(fd, buffer); !er || er.get() != hsize ||
       std::memcmp(&buffer[0], &header[0], hsize) != 0) {
```

Comment:
I think this would be more readable if the declaration of `er` and the `if` were separated.